tests/python/pants_test/engine/test_round_engine.py

```diff
@@ -215,6 +218,19 @@ def test_cli_goals_deduped(self):
                         self.as_goals('goal1', 'goal2', 'goal1', 'goal3', 'goal2'))
     self.assert_actions('task1', 'task2', 'task3')
 
+  def test_task_subclass_singletons(self):
+    # Install the same task class twice (before/after Goal.clear()) and confirm that the
+    # resulting task is equal.
+    class MyTask(Task):
+      pass
+    def install():
+      reg = super(RoundEngineTest, self).install_task(name='task1', action=MyTask, goal='goal1')
+      return reg.task_types()
+    task1_pre, = install()
+    Goal.clear()
+    task1_post, = install()
+    self.assertEquals(task1_pre, task1_post)
```

Comment:
Do you want to check that they have the same id? Or is semantic equality all we actually care about anyway?

Reply:
Semantic equality is really what matters, yea. Although in theory I could/should have checked `__hash__` as well.